### ISSUE TITLE
Is there any way to completely override the sorting logic? [Help] 

### ISSUE THREAD

**Issue by ashndev** [bosskmk/pluto_grid#617]
Hi There!

Couple of questions: 

1. Is it possible to completely override the sorting logic? What I want to be able to do is click on the header cell and get sorted records from the server without applying the Pluto sorting. I still want to be able to update the icon though to indicate the sorting order. 
2. Is there a way to have separate icons for the sort indication and the pop-up menu? Right now the menu icon changes to a sorting icon when you sort, but I want to be able to still see the menu icon separately even when sort is applied.

**Comment by bosskmk** [comment]
I don't know if I understood your question well.

1. The sort is handled by the server
> `PlutoLazyPagination` or `PlutoInfinityScrollRows` plugins allow sorting to be handled on the server. In addition, the above two plugins allow the user to refer to the code and create a new widget with the desired appearance.
https://pluto.weblaze.dev/rows-from-server-side
As you can see from the above document, the `fetchWithSorting` property handles sorting on the server.

2. Separating the sort icon and menu icon
> This is not possible now. But let's consider an update.

**Comment by bosskmk** [comment]
Feature. separating sort and menu icons.

**Comment by ashndev** [comment]
Hi @bosskmk, ahhh.. I see, but with the InfinityScrollRows the fetch will only happen when you reach the bottom of the screen? 

I am trying to implement the grid into an already existing app. So, I already have a ScrollController which handles the data fetching when you reach a certain part of the screen and it is passed down from another widget to the widget where the Pluto grid is located. That widget where the ScrollController is initialized also provides the widget where the Pluto grid is located with the updated data. So all I need to do with the Pluto grid, when I get some new data is to just reset the rows to the new set of rows. I've got that already handled and it works, but now the problem is with sorting and filtering. 

With the above-mentioned setup, when I click on the column header to sort I want to be able to call my own method that will trigger data fetching with the specified sort and the same for filtering when I set a filter. But I guess that kind of thing is currently only possible with the PlutoInifinityScrollRows, right?

Sorry if this is all a bit confusing, but I hope it kind of makes sense.

**Comment by bosskmk** [comment]
I can't recommend a good way because I don't know your business logic, but I think you can implement the behavior you want by doing the following.


```dart
import 'package:flutter/material.dart';
import 'package:pluto_grid/pluto_grid.dart';

import '../dummy_data/development.dart';

class EmptyScreen extends StatefulWidget {
  static const routeName = 'empty';

  const EmptyScreen({Key? key}) : super(key: key);

  @override
  _EmptyScreenState createState() => _EmptyScreenState();
}

class _EmptyScreenState extends State<EmptyScreen> {
  late List<PlutoColumn> columns;

  late List<PlutoRow> rows;

  late PlutoGridStateManager stateManager;

  @override
  void initState() {
    super.initState();

    columns = [
      PlutoColumn(
        title: 'column1',
        field: 'column1',
        type: PlutoColumnType.text(),
      ),
      PlutoColumn(
        title: 'column2',
        field: 'column2',
        type: PlutoColumnType.text(),
      ),
      PlutoColumn(
        title: 'column3',
        field: 'column3',
        type: PlutoColumnType.text(),
      ),
    ];

    rows = DummyData.rowsByColumns(length: 10, columns: columns);
  }

  void _handleOnSorted(PlutoGridOnSortedEvent event) {
    // Do something.
    print(event);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Container(
          padding: const EdgeInsets.all(15),
          child: PlutoGrid(
            columns: columns,
            rows: rows,
            onChanged: (PlutoGridOnChangedEvent event) {
              print(event);
            },
            onLoaded: (PlutoGridOnLoadedEvent event) {
              stateManager = event.stateManager;
              stateManager.setSortOnlyEvent(true);
            },
            onSorted: _handleOnSorted,
            configuration: const PlutoGridConfiguration(),
          ),
        ),
      ),
    );
  }
}
```

**Comment by bosskmk** [comment]
If you do the above, the sort icon will change, but it won't do anything.
You can handle the data after sorting by implementing the `_handleOnSorted` method.

**Comment by ashndev** [comment]
Great, thank you! This might just work :)

**Comment by bosskmk** [comment]
@ashndev 
Added filtering related code.

**Comment by bosskmk** [comment]
For more information about event.filterRows , please refer to the link below.
https://github.com/bosskmk/pluto_grid/blob/master/lib/src/plugin/pluto_infinity_scroll_rows.dart#L26

**Comment by ashndev** [comment]
Thank you!

**Comment by ashndev** [comment]
Hi @bosskmk the sorting worked beautifully, thank you! I am just having a bit of a problem with filtering. 

I show the filter inputs below the column header and by default they are set to "Contains", then when I go to change the type of filter to "Ends with" for example with the filter pop-up UI, the input below the column header doesn't update. Am I missing a step somewhere? I followed the examples you provided. 

<img width="291" alt="Screen Shot 2022-10-27 at 10 07 45 AM" src="https://user-images.githubusercontent.com/87834445/198355855-f2430cf0-f75b-4e38-8c2a-d1b96c1f4845.png">
<img width="609" alt="Screen Shot 2022-10-27 at 10 08 24 AM" src="https://user-images.githubusercontent.com/87834445/198355869-f56e7a79-f4d2-43e0-be55-8211b61c7a46.png">

**Comment by ashndev** [comment]
Gotcha, thank you! And is there a way to get the filter icon to show up when I set the filter manually as I am doing? Because when I set this to true `plutoDataGridSource!.stateManager!.setFilterOnlyEvent(true)` that filter icon doesn't show up. 
<img width="113" alt="Screen Shot 2022-10-27 at 11 30 38 AM" src="https://user-images.githubusercontent.com/87834445/198370416-704df886-dd39-47c2-a163-9d0cc9911b4c.png">
